src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -194,4 +196,19 @@ public ResponseEntity<ZonedDateTime> getIrisAcceptedForStudent() {
         User user = userRepository.getUser();
         return ResponseEntity.ok().body(user.getIrisAcceptedTimestamp());
     }
+
+    /**
+     * PUT users/sshpublickey : sets the ssh public key
+     *
+     * @return the ResponseEntity with status 200 (OK), with status 404 (Not Found), or with status 400 (Bad Request)
+     */
+    @PutMapping("users/sshpublickey")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Void> addSshPublicKey(String sshPublicKey) throws NoSuchAlgorithmException {
```

Comment:
This endpoint won't work without `@RequestBody`.
```suggestion
    public ResponseEntity<Void> addSshPublicKey(@RequestBody @NotNull String sshPublicKey) throws NoSuchAlgorithmException {
```